gadget/mountedfilesystem.go

```diff
@@ -672,21 +678,28 @@ func (f *mountedFilesystemUpdater) checkpointPrefix(dstRoot, target string, back
 }
 
 func (f *mountedFilesystemUpdater) observedBackupOrCheckpointFile(dstRoot, source, target string, preserveInDst []string, backupDir string) error {
-	willBeWritten, err := f.backupOrCheckpointFile(dstRoot, source, target, preserveInDst, backupDir)
+	willBeWritten, originalDataPath, err := f.backupOrCheckpointFile(dstRoot, source, target, preserveInDst, backupDir)
 	if err != nil {
 		return err
 	}
 	if willBeWritten {
 		srcPath := f.entrySourcePath(source)
 		dstPath, _ := f.entryDestPaths(dstRoot, source, target, backupDir)
-		if err := observe(f.updateObserver, ContentUpdate, f.ps, f.mountPoint, srcPath, dstPath); err != nil {
+		data := &ContentChange{
+			// content of the new data
+			After: srcPath,
+			// this is the original data that was present before the
+			// update or "" if the file did not exist
+			Before: originalDataPath,
+		}
+		if err := observe(f.updateObserver, ContentUpdate, f.ps, f.mountPoint, dstPath, data); err != nil {
 			return fmt.Errorf("cannot observe pending file write %v\n", err)
 		}
 	}
 	return nil
 }
 
-func (f *mountedFilesystemUpdater) backupOrCheckpointFile(dstRoot, source, target string, preserveInDst []string, backupDir string) (willWrite bool, err error) {
+func (f *mountedFilesystemUpdater) backupOrCheckpointFile(dstRoot, source, target string, preserveInDst []string, backupDir string) (willWrite bool, originalDataPath string, err error) {
```

Comment:
at this point this probably merits a comment about the meaning of the return values